packages/components/src/Tokens/TokensAccountBuy.tsx

```diff
@@ -33,92 +33,77 @@ export interface TokenAccountBuyProps {
   network: string;
   faqUrl: string;
   step?: string;
+  onBuyComplete(): void;
 }
 
-export interface TokenAccountBuyStates {
-  step?: string;
-}
+export const UserTokenAccountBuy: React.StatelessComponent<TokenAccountBuyProps> = props => {
+  const { foundationAddress, network, faqUrl, step, onBuyComplete } = props;
+  let tokenSection;
 
-export class UserTokenAccountBuy extends React.Component<TokenAccountBuyProps, TokenAccountBuyStates> {
-  constructor(props: any) {
-    super(props);
-    this.state = { step: this.props.step };
-  }
+  if (step === TOKEN_PROGRESS.DISABLED) {
```

Comment:
I think this is fine for now, but we should break these large render functions down into smaller ones, it'll be easier to read.

Reply:
Agreed, this will need a refactor when I add the sell flow. I will break down the components in that ticket.